Set custom config via config attribute

waterkip opened the pull request:
Bon dia Jose,

I have a patch that enables us (me and my employer) to use different kubectl configs. We have several clouds and therefore use `kubectl --kubeconfig=/path/to/kubeconfig`. The first commit deals with that.

I've also taken the liberty to add some predicates, so `$foo if defined $self->foo` becomes `$foo if $self->has_foo`.

And I've changed `diag` to `note` in the testsuite, diag always shows diagnostics, note only when running `prove -v`.

Thanks for the module, I like it!

Mucho gracias :) 
Wesley

waterkip committed:
Change diag to note in test

note is shown with prove -lv, diag is always shown.

waterkip commented:
I saw that I didn't update the documenation. Added two fixups which can be squashed (git rebase -i --autosquash).

pplu commented:
Hi!

  Thanks for your contribution! (BTW: this module was meant as a stop-gap solution before building a Perl REST Client for Kubernetes. Tell me if you would be interested to participate building it).

pplu commented on lib/Kubectl/CLIWrapper.pm:
```diff
@@ -6,27 +6,36 @@ package Kubectl::CLIWrapper {
+    $options{username}   = $self->username  if $self->has_username;
+    $options{password}   = $self->password  if $self->has_password;
+    $options{namespace}  = $self->namespace if $self->has_namespace;
+    $options{kubeconfig} = $self->config    if $self->has_config;
```
I'm not sure of changing the name of the kubeconfig command line option for config. I did that with the insecure_tls option, and I regret it (since it's more intuitive that these options align). Can you please change the name, and document the option together with username, password, etc?

waterkip replied:
Done :+1:

pplu commented on t/02_interface.t:
```diff
@@ -9,7 +9,7 @@ use Kubectl::CLIWrapper;
   my $control = Kubectl::CLIWrapper->new(namespace => 'ns1');
   my $command = join ' ', $control->command_for('create', 'pod');
-  diag $command;
+  note $command;
```
Thanks for making me notice "note". I only knew "diag" 👍

waterkip replied:
Cool :)

waterkip committed:
Set custom config via config attribute

Being able to use different kubeconfig's is a nice perk to have.

waterkip committed:
Refactor code to use a predicate.

This refactor also removes the namespace default, as this is already
handled by kubectl and may or may not interfere with
--all-namespaces=true.

waterkip commented:
I'm happy to help where needed.

pplu commented:
Released to CPAN: version 0.03
